README.md

```diff
@@ -312,44 +309,58 @@ metadata:
 spec:
   description: string # optional up to 1050 characters
   thresholdMetric: # either thresholdMetric or ratioMetric must be provided
-    metricSource:
-      metricSourceRef: string # optional, this field can be used to refer to DataSource object
-      type: string # optional, this field describes predefined metric source type e.g. Prometheus, Datadog, etc.
-      spec:
-        # arbitrary chosen fields for every data source type to make it comfortable to use
-        # anything that is valid YAML can be put here.
+    # either dataSourceRef or <<dataSourceName>> must be provided
+    dataSourceRef: string # refer to already defined DataSource object
+    <<dataSourceName>>: # inline whole DataSource e.g. cloudWatch, datadog, prometheus (arbitrary chosen, implementor decision)
+      # fields used for creating a connection with particular datasource e.g. AccessKeys, SecretKeys, etc.
+      # everything that is valid YAML can be put here
+    spec:
```

Comment:
Should the usage of `spec` here be renamed as it doesn't equate to a spec of a kind, as it does in other parts of the spec?

Reply:
We can think about it. The most obvious word is `query`, but not every data source uses such naming as described in https://github.com/OpenSLO/OpenSLO/issues/107. Have you any idea? 

Having `spec` in many places is something common in K8s, e.g. in Deployment
```yml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
spec:
  selector:
    matchLabels:
      app: nginx
  replicas: 2 # tells deployment to run 2 pods matching the template
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.14.2
        ports:
        - containerPort: 80
```